crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1845,6 +1844,22 @@ impl Room {
             .power_levels())
     }
 
+    /// Gets the suggested role for the user with the provided `user_id`.
+    /// This method checks the `RoomPowerLevels` events instead of loading the
+    /// member list and looking for the member.
+    pub async fn get_suggested_user_role(&self, user_id: OwnedUserId) -> Result<RoomMemberRole> {
```

Comment:
nit: in public APIs, prefer using `&UserId` over `OwnedUserId`, unless you actually need the ownership. It can avoid cloning at the call sites.

Reply:
Done in https://github.com/matrix-org/matrix-rust-sdk/pull/3170/commits/307e5bf0333c5b45be4963efaa9ef5ff2c09902d, same for the other case.